stan/math/opencl/err/check_diagonal_zeros.hpp

```diff
@@ -25,14 +28,11 @@ inline void check_diagonal_zeros(const char* function, const char* name,
   cl::Context ctx = opencl_context.context();
   try {
     int zero_on_diagonal_flag = 0;
-    cl::Buffer buffer_flag(ctx, CL_MEM_READ_WRITE, sizeof(int));
-    cmd_queue.enqueueWriteBuffer(buffer_flag, CL_TRUE, 0, sizeof(int),
-                                 &zero_on_diagonal_flag);
-    opencl_kernels::check_diagonal_zeros(cl::NDRange(y.rows(), y.cols()),
-                                         y.buffer(), buffer_flag, y.rows(),
-                                         y.cols());
-    cmd_queue.enqueueReadBuffer(buffer_flag, CL_TRUE, 0, sizeof(int),
-                                &zero_on_diagonal_flag);
+    matrix_cl zeros_flag(1, 1);
+    copy(zeros_flag, zero_on_diagonal_flag);  // NOLINT
```

Comment:
Is there a different standard name we can use? Or alt can we remove this option from the linter?

Reply:
(again not for this PR but general discussion) idk worst case we can always use copy_cl or cl_copy (and it might not be a crazy idea to have different names for the to GPU vs. from GPU options). but we can make an issue about this and think about it a little more there.

Reply:
(I'll put this on my to-do list)

I'm fine with changing the names, though will want to shoot the breeze so we all think it's reasonable. Is `to_cl` and `from_cl` too much? I'm also fine with `copy_cl`

Reply:
I think those are probably fine. Anything that doesn't require //NOLINT at every call site 😂